grammars/css.cson

```diff
@@ -528,251 +1632,363 @@
-            'name': 'string.quoted.single.css'
-            'patterns': [
-              {
-                'match': "(?<=['\\s])[a-zA-Z*]+(-[a-zA-Z0-9*]*+)*+(?=['\\s])"
-                'name': 'support.constant.language-range.css'
-              }
-            ]
+            'captures':
+              '1':
+                'name': 'storage.modifier.ignore-case.css'
+            'match': '(?<=["\'\\s]|^|\\*/)\\s*([iI])\\s*(?=[\\s\\]]|$|/\\*)'
+          }
+          {
+            'captures':
+              '1':
+                'name': 'string.unquoted.attribute-value.css'
+                'patterns': [
+                  'include': '#escapes'
+                ]
+            'match': '(?x)(?<==)\\s*((?!/\\*)(?:[^\\\\"\'\\s\\]]|\\\\.)+)'
+          }
+          {
+            'include': '#escapes'
+          }
+          {
+            'match': '[~|^$*]?='
+            'name': 'keyword.operator.pattern.css'
           }
           {
-            'match': ','
+            'match': '\\|'
             'name': 'punctuation.separator.css'
           }
+          {
+            'captures':
+              '1':
+                'name': 'entity.other.namespace-prefix.css'
+                'patterns': [
+                  {
+                    'include': '#escapes'
+                  }
+                ]
+            'match': '''(?x)
+              # Qualified namespace prefix
+              ( -?(?!\\d)(?:[\\w-]|[^\\x00-\\x7F]|\\\\(?:[0-9a-fA-F]{1,6}|.))+
+              | \\*
+              )
+              # Lookahead to ensure there's a valid identifier ahead
+              (?=
+                \\| (?!\\s|=|$|\\])
+                (?: -?(?!\\d)
+                 |   [\\\\\\w-]
+                 |   [^\\x00-\\x7F]
+                 )
+              )
+            '''
+          }
+          {
+            'captures':
+              '1':
+                'name': 'entity.other.attribute-name.css'
+                'patterns': [
+                  'include': '#escapes'
+                ]
+            'match': '''(?x)
+              (-?(?!\\d)(?:[\\w-]|[^\\x00-\\x7F]|\\\\(?:[0-9a-fA-F]{1,6}|.))+)
+              \\s*
+              (?=[~|^\\]$*=]|/\\*)
+            '''
+          }
         ]
       }
       {
-        'begin': '((:)not)(\\()'
+        'include': '#pseudo-classes'
+      }
+      {
+        'include': '#pseudo-elements'
+      }
+      {
+        'include': '#functional-pseudo-classes'
+      }
+      # Custom HTML elements
+      {
+        'match': '''(?x) (?<![@\\w-])
+          (?=            # Custom element names must:
+            [a-z]        # - start with a lowercase ASCII letter,
+            \\w* -       # - contain at least one dash
+          )
+          (?:
+            (?![A-Z])    # No uppercase ASCII letters are allowed
+            [\\w-]       # Allow any other word character or dash
+          )+
+          (?![(\\w-])
+        '''
+        'name': 'entity.name.tag.custom.css'
+      }
+    ]
+  'string':
```

Comment:
This is an optional change, but most of the bundled languages use `strings` instead of `string`.